modules/core/include/opencv2/core/ocl.hpp

```diff
@@ -276,6 +276,31 @@ class CV_EXPORTS Platform
     Impl* p;
 };
 
+/*
+// Attaches OpenCL context to OpenCV
+// Params:
+//   platformName - name of OpenCL platform, must match with platform OpenCV initialized for
+//   platfromID   - ID of platform attached context was created for
+//   context      - attached OpenCL context
+//   deviceID     - ID of device, must be created from attached context
+*/
+CV_EXPORTS void attachContext(String& platformName, void* platformID, void* context, void* deviceID);
+
+/*
+// Convert OpenCL clBuffer to UMat
+// Note:
+//   buffer should represent 2D image, compatible with OpenCV. Memory is not copied from clBuffer to UMat
+// Params:
+//   rows - number of rows
+//   cols - number of cols
+//   type - OpenCV type of image
+//   cl_mem_obj - source clBuffer handle
+//   dst  - destination UMat
+//   usageFlags - OpenCV UMatUsageFlags
+*/
+CV_EXPORTS void convertFromBuffer(int rows, int cols, int type, void* cl_mem_obj,
```

Comment:
rows & cols can be retrieved using clGetImageInfo() and should better be retrieved this way